reindex.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/olivere/elastic.v5"
+)
+
+var (
+	ErrTooManyResults = errors.New("Too many results in prefix")
+)
+
+func logTime(prefixlog *logrus.Entry, start time.Time, rows *struct {
+	rows                int64
+	documents           int64
+	processed           int64
+	dataobjects         int64
+	dataobjects_added   int64
+	dataobjects_updated int64
+	dataobjects_removed int64
+	colls               int64
+	colls_added         int64
+	colls_updated       int64
+	colls_removed       int64
+}) {
+	prefixlog.Infof("Processed %d entries (%d rows, %d documents, processed %d data objects (+%d,U%d,-%d), %d colls (+%d,U%d,-%d)) in %s", rows.processed, rows.rows, rows.documents, rows.dataobjects, rows.dataobjects_added, rows.dataobjects_updated, rows.dataobjects_removed, rows.colls, rows.colls_added, rows.colls_updated, rows.colls_removed, time.Since(start).String())
+}
+
+func ReindexPrefix(db *ICATConnection, es *ESConnection, prefix string) error {
```

Comment:
This should be broken up into multiple functions to make it easier to visualize the algorithm.

Reply:
Yeah, another good idea. My own fault, since I effectively grew this from an empty function to... well, this, without sitting down to think about the structure. I'll give it a look.